bokehjs/src/lib/models/text/utils.ts

```diff
@@ -0,0 +1,31 @@
+import {isString} from "core/util/types"
+import {TeX} from "models/text/index"
+import {BaseText} from "./base_text"
+
+export function is_tex_string(text: unknown): text is string {
+  if (!isString(text)) return false
+
+  if (text.startsWith("$$") && text.endsWith("$$"))
+    return true
+
+  else if (text.startsWith("\\[") && text.endsWith("\\]"))
+    return true
+
+  else if (text.startsWith("\\(") && text.endsWith("\\)"))
+    return true
+
+  return false
+};
+
+export function tex_from_text_like(text: string | BaseText): TeX | null {
+  let math_text: TeX = new TeX()
+
+  if (text instanceof TeX)
+    math_text = text
+  else if (is_tex_string(text))
+    math_text.text = text.slice(2, -2)
+  else
+    return null
+
+  return math_text
+}
```

Comment:
please double check but I think this is equivalent and does not require a `let` (also the last return above seem unreachable)

```suggestion
export function tex_from_text_like(text: string | BaseText): TeX | null {
  if (text instanceof TeX)
    return text
  if (is_tex_string(text))
    return new Tex({text: text.slice(2, -2)})
  return null
}
```